zerver/views/user_settings.py

```diff
@@ -271,6 +272,12 @@ def json_change_settings(
     if len(request_notes.ignored_parameters) > 0:
         result["ignored_parameters_unsupported"] = list(request_notes.ignored_parameters)
 
+    if enable_drafts_syncing is not None:
+        user_profile.enable_drafts_synchronization = enable_drafts_syncing
+        user_profile.save(update_fields=["enable_drafts_synchronization"])
+        if enable_drafts_syncing is False:
+            Draft.objects.filter(user_profile=user_profile).delete()
```

Comment:
I think we want the API name and database name to match; so `enable_drafts_synchronization != enable_drafts_syncing` is a bug.  I'd probably do `enable_drafts_synchronization` as the one to keep.

Reply:
We can likely also use the `property_types` system for this property; then you'd just need to move the `Draft.objects.filter` line to the `do_change_display_setting`  function in `actions.py`.

Reply:
All of the properties in `UserBaseSettings.property_types` seem to be display related properties, so I'm against putting this there. Are you sure you want to move it?

Reply:
Yes, see https://chat.zulip.org/#narrow/stream/378-api-design/topic/settings.20endpoints/near/1237310.  Basically, (1) `display_settings` had become the catch-all place for user settings that were not notification settings and (2) we're partway through merging these into a single "settings" object, so that the API doesn't depend on this sort of detail.

Reply:
I couldn't find any method named `do_change_display_setting`, did you mean `do_set_user_display_setting`?